File: Java/src/main/java/com/targeter/server/entity/Target.java
package com.targeter.server.entity;

import com.vladmihalcea.hibernate.type.basic.PostgreSQLEnumType;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.Type;
import org.hibernate.annotations.TypeDef;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.persistence.*;

@Getter
@Setter
@Entity
@Table(name = "target")
@TypeDef(
    name = "pgsql_enum",
    typeClass = PostgreSQLEnumType.class
)
public class Target {
  @Id
  @Column(name = "id")
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @ManyToOne
  @JoinColumn(name = "user_id")
  private @NotNull User user;

  @Column(name = "text")
  private @NotNull String text;

    @ManyToOne
    @JoinColumn(name = "target_category_id")
    private @NotNull TargetCategory category;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", columnDefinition = "status")
    @Type(type = "pgsql_enum")
    private @NotNull Status status;

    /*@OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "schedule", referencedColumnName = "id")*/
    @OneToOne
    @JoinColumn(name = "schedule")
    private @Nullable Schedule schedule;

    @Column(name = "photo_report")
    private @Nullable byte[] photoReport;
}

File: motivator/motivator/Models/Target.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace motivator.Models
{
    public partial class Target
    {
        public long Id { get; set; }
        public long UserId { get; set; }
        public string Text { get; set; }
        public long TargetCategoryId { get; set; }
        public long? Schedule { get; set; }
        public byte[] PhotoReport { get; set; }

        public virtual Schedule ScheduleNavigation { get; set; }
        public virtual TargetCategory TargetCategory { get; set; }
        public virtual TargeterUser User { get; set; }
    }
}
